kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1124 @@
+                            auto p_node = r_geom(i_node);
+                            auto& ext_op_data = ext_op_map[p_node];
+
+                            // Loop the current negative node to get its extrapolation operator data and apply the weight to make the basis conformant
+                            // Note that we need to check for the ids to match in the geometry as nodes in the map are mixed
+                            for (auto it_data = ext_op_data.begin(); it_data != ext_op_data.end(); ++it_data) {
+                                auto& r_node_data = *it_data;
+                                std::size_t data_node_id = (std::get<0>(r_node_data))->Id();
+                                for (std::size_t i_cl = 0; i_cl < n_cl_nodes; ++i_cl) {
+                                    auto& p_cl_node = cloud_nodes_vector(i_cl);
+                                    if (p_cl_node->Id() == data_node_id) {
+                                        const double i_cl_node_N = std::get<1>(r_node_data);
+                                        N_container(i_cl) += i_node_N * i_cl_node_N;
+                                        for (std::size_t d = 0; d < n_dim; ++d) {
+                                            DN_DX_container(i_cl,d) += i_node_grad_N(d) * i_cl_node_N;
+                                        }
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    // Create a new condition with a geometry made up with the basis nodes
+                    auto p_prop = rElement.pGetProperties();
+                    auto p_cond = mpConditionPrototype->Create(++max_cond_id, cloud_nodes_vector, p_prop);
+                    p_cond->Set(ACTIVE, true);
+                    mpBoundarySubModelPart->AddCondition(p_cond);
+
+                    // Store the SBM BC data in the condition database
+                    p_cond->SetValue(ELEMENT_H, h);
+                    const double n_norm = norm_2(pos_int_n[i_g]);
+                    p_cond->SetValue(NORMAL, pos_int_n[i_g] / n_norm);
+                    p_cond->SetValue(INTEGRATION_WEIGHT, pos_int_w[i_g]);
+                    p_cond->SetValue(INTEGRATION_COORDINATES, i_g_coords);
+                    p_cond->SetValue(SHAPE_FUNCTIONS_VECTOR, N_container);
+                    p_cond->SetValue(SHAPE_FUNCTIONS_GRADIENT_MATRIX, DN_DX_container);
+                }
+            }
+        }
+    }
+
+    void ShiftedBoundaryMeshlessInterfaceUtility::CalculateMeshlessBasedNonConformingExtensionBasis()
+    {
+        // Set the required interface flags
+        SetInterfaceFlags();
+
+        // Set the modified shape functions factory
+        // Note that unique geometry in the mesh is assumed
+        const auto& r_begin_geom = mpModelPart->ElementsBegin()->GetGeometry();

```

Comment:
boom if the modelpart has no elements

Reply:
Nice catch. Now an error is thrown (also in the other places this is done).